src/crafting_gui.cpp

```diff
@@ -476,11 +476,124 @@ const recipe *select_crafting_recipe( int &batch_size_out )
             item_info_scroll = 0;
             item_info_scroll_popup = 0;
         }
-        std::vector<iteminfo> info;
-        item_info_cache.dummy.info( true, info, count );
-        item_info_data data( item_info_cache.dummy.tname( count ),
-                             item_info_cache.dummy.type_name( count ),
-                             info, {}, scroll_pos );
+        std::vector<iteminfo> info; //This is the information that will be printed later
+
+        //getting item info clears the vector, so to append multiple results, we need a temporary vector
+        std::vector<iteminfo> temp_info;
+        /*We need to do some calculations to put together the results summary and very similar calculations to
+         put together the details, so, have a separate vector specifically for the details, to be appended later */
+        std::vector<iteminfo> details_info;
+
+        const auto insert_iteminfo_separator_line = [&]( std::vector<iteminfo> &info_vec ) {
+            /*Calculation is the current method for calculating the width of the iteminfo panel.  TODO: Link it back
+              to the previously calculated value so that this won't break if the calculation changes. */
+            info_vec.emplace_back( "DESCRIPTION", std::string( std::min( TERMX,
+                                   FULL_SCREEN_WIDTH * 2 ) - FULL_SCREEN_WIDTH - 1, '-' ) );
+        };
+        const auto insert_iteminfo_blank_line = [&]( std::vector<iteminfo> &info_vec ) {
+            info_vec.emplace_back( "DESCRIPTION", "--" );
+        };
+
+        /*Temporary items to properly handle names/quantities, used in conjunction with item_info_cache.dummy
+        TODO: It may be beneficial to move these into item_info_cache.  However, given the complexity of
+        putting the iteminfo together, it would also be necessary to cache "info", and that might not be as
+        desirable. */
+        item dummy_result;
+        item dummy_container;
+
+        //Set up summary at top so people know they can look further to learn about byproducts and such
+        //First, see if we need it at all:
+        if( rec->container_id() == itype_id::NULL_ID() && !rec->has_byproducts() ) {
+            //We don't need a summary for a single item, just give us the details
+            info.emplace_back( "DESCRIPTION",
+                               _( "<bold>Result: </bold>" + item_info_cache.dummy.display_name( count ) ) );
+            item_info_cache.dummy.info( true, temp_info, count );
+            info.insert( std::end( info ), std::begin( temp_info ), std::end( temp_info ) );
+        } else { //We do need a summary
```

Comment:
Couldn't this whole else branch be made into a function? It's quite large with it's 180 lines and the tabulation is getting ridiculous.

Also note the error: `function 'select_crafting_recipe' exceeds recommended size/complexity thresholds [readability-function-size,-warnings-as-errors]`

You will have to split this function into smaller ones, as with your addition you exceed the allowed function line count.

Reply:
I've split the entire recipe info calculation out of select_crafting_recipe(), then split up the function a bit.  Hopefully the new arrangement is a bit easier to follow.
select_crafting_recipe() is still... large, but it's a little shorter than before.  Further changes should probably be separate PRs.